src/main/java/com/microsoft/sqlserver/jdbc/SharedTimer.java

```diff
@@ -16,10 +17,11 @@
      * Always update serialVersionUID when prompted
      */
     private static final long serialVersionUID = -4069361613863955760L;
-    
+
     static final String CORE_THREAD_PREFIX = "mssql-jdbc-shared-timer-core-";
     private static final AtomicLong CORE_THREAD_COUNTER = new AtomicLong();
-    private static SharedTimer instance;
+    private static volatile SharedTimer instance;
+    private static Object lock = new Object();
```

Comment:
This should be final as it must never change.

Could also reorder it above the SharedTimer so that the final fields come before all non-final fields.